src/main/java/seedu/address/model/jobs/DeliveryJob.java

```diff
@@ -106,8 +106,12 @@ public boolean getDeliveredStatus() {
         return isDelivered;
     }
 
-    public LocalDate getDeliverDate() {
-        return LocalDate.now();
+    public LocalDate getDate() throws NoSuchElementException {
+        return deliveryDate.get().getDate();
+    }
```

Comment:
it would be better if your fontend calls optional methods to handle empty() cases instead of handling it as an error in exception clause.